main.go

```diff
@@ -12,17 +12,38 @@ import (
 	"reflect"
 )
 
-var socket = flag.String("socket", "", "Socket to listen into")
+var dump = flag.String("dump", "", "Dump info about `plugin` as a MessagePack object")
+var socket = flag.String("socket", "", "Starts a server listening on the given socket `path`.")
+var pluginsDir = flag.String("dir", "", "Set directory `path` where to search plugins")
 
 func init() {
 	flag.Parse()
 
-	if *socket == "" {
+	if *socket == "" && *dump == "" {
 		flag.Usage()
 		os.Exit(2)
 	}
 }
 
+func dumpInfo() {
+	s := newServer()
+
+	info := PluginInfo{}
+	err := s.GetPluginInfo(*dump, &info)
+	if err != nil {
+		log.Printf("%s", err)
+	}
+
+	var handle codec.MsgpackHandle
```

Comment:
Is this chunk starting here a leftover? Doesn't seem like the result is being used or returned anywhere

Reply:
the `handle` is used for the call to `.NewEncoder(os.Stdout, &handle)`  this is the way to specify encoding options.

Reply:
Oops, I missed the stdout in there, sorry!